src/Rings/FreeAssAlgIdeal.jl

```diff
@@ -92,15 +82,40 @@ end
 Return `true` if calculations with intermediate degrees bounded by `deg_bound`
 prove that $a$ is in $I$. Otherwise, a return of `false` indicates an
 inconclusive answer, but larger `deg_bound`s give more confidence in a negative
-answer.
+answer. If `deg_bound` is not provided, the default value is `-1`, which means 
+that no degree bound is imposed, which leads to a computation that uses a much
+slower algorithm, that may not terminate, but returns a full groebner basis if
+```jldoctest
+free, (x,y,z) = free_associative_algebra(QQ, ["x", "y", "z"])
+f1 = x*y + y*z
+I = ideal([f1])
+ideal_membership(f1, I, 4)
+
+# output
+true
+```
 """
-function ideal_membership(a::FreeAssAlgElem, I::FreeAssAlgIdeal, deg_bound::Int)
+function ideal_membership(a::FreeAssAlgElem, I::FreeAssAlgIdeal, deg_bound::Int=-1)
+  return ideal_membership(a, I.gens, deg_bound)
```

Comment:
try and use functions insteads of accessing properties
```suggestion
  return ideal_membership(a, gens(I), deg_bound)
```